controller.go

```diff
@@ -284,13 +284,17 @@ func syncPV(pv *PV) {
 			// NOTE: releasePV may either release the PV back into the pool or
 			// recycle it or do nothing (retain)
 
-			// HOWTO RELEASE A PV
-			pv.Status.Phase = Released
-			if err := CommitPVStatus(pv); err != nil {
-				// Status was not saved; we will fall back into the same
-				// condition in the next call to this method
-				return
+			// Make sure we don't overwrite previous failure of
+			// recycler/deleter.
+			if pv.Status.Phase != Failed {
```

Comment:
`PV.Status.Phase` is explicitly set to `Failed` here: https://github.com/pmorie/pv-haxxz/blob/master/controller.go#L331

It's not in our pseudocode, but I set it to Failed also every time that recycling fails, e.g. steps 2.5 and 4 (pod fails) in the recycling process at lines https://github.com/pmorie/pv-haxxz/blob/master/controller.go#L318